contentcuration/contentcuration/static/js/edit_channel/channel_set/views/ChannelSetModal.vue

```diff
@@ -0,0 +1,99 @@
+<template>
+
+  <div ref="topmodal" class="modal fade" tabindex="-1">
+    <div class="modal-dialog">
+      <div class="modal-content modal-dialog-default" id="channel_set_modal_content">
+        <div class="modal-header">
+          <h4 class="modal-title">{{ modalTitle }}</h4>
+        </div>
+        <div class="modal-body">
+          <ChannelSetDialog>
+            <component :is="pageType" />
+          </ChannelSetDialog>
+        </div>
+      </div>
+    </div>
+  </div>
+
+</template>
+
+
+<script>
+
+import { mapGetters, mapActions } from 'vuex';
+import { PageTypes } from '../constants';
+import ChannelSetDialog from './ChannelSetDialog.vue';
+import ChannelSetList from './ChannelSetList.vue';
+import ChannelSelectView from './ChannelSelectView.vue';
+
+const pageNameToComponentMap = {
+  [PageTypes.SELECT_CHANNELS]: 'ChannelSelectView',
+  [PageTypes.VIEW_CHANNELS]: 'ChannelSetList',
+};
+
+export default {
+  name: 'ChannelSetModal',
+  $trs: {
+    newSetHeader: 'New Collection',
+    editingSetHeader: 'Editing Collection',
+  },
+  components: {
+    ChannelSetDialog,
+    ChannelSetList,
+    ChannelSelectView,
+  },
+  mounted() {
+    this.openModal();
+  },
+  computed: Object.assign(
+    mapGetters('channel_set', [
+      'isNewSet',
+      'currentPage'
+    ]),
+    {
+      pageType() {
+        return pageNameToComponentMap[this.currentPage];
+      },
+      modalTitle() {
+        if (this.isNewSet) {
+          return this.$tr('newSetHeader');
+        }
+        return this.$tr('editingSetHeader');
+      },
+    }
+  ),
+  methods: Object.assign(
+    {
+      openModal() {
+        $(this.$refs.topmodal)
```

Comment:
I am not a huge fan of using jQuery inside Vue - I assume this is to leverage the bootstrap modal jQuery plugin?

I assume the main advantage of this is to execute the closing animation - this might be a good opportunity to experiment with vue transitions, and you can have an internal data attribute for the modal component that defines closed vs open, then use Vue transitions to define the animation https://vuejs.org/v2/guide/transitions.html

Reply:
Yes, it's to use bootstrap. However, I did find this library- would it make sense to import this? https://bootstrap-vue.js.org/docs

Reply:
There are also some listeners that catch when the modal is closing (triggers check for unsaved changes) and when it has been closed (triggers destroy method). There are also event listeners like clicking outside of the modal or pressing the escape button that trigger closing as well. It might take a lot longer to implement a vue modal with all the functionalities of bootstrap's modals, unless there's a vue modal library that does the same thing?

Reply:
Vue's got some mechanisms for each of these that lives in any component, not just modals. But YMMV!

> There are also some listeners that catch when the modal is closing (triggers check for unsaved changes) and when it has been closed (triggers destroy method)
If the parent is what controls the opening/closing of a child modal via a `v-if`, it's already in full control of the display logic and doesn't need the modal to tell it when the child modal is opened or closed. 
If you _just_ go the `v-if route`, you could also use pretty much the same event-based structure you're using here (where the parent waits for the modal to tell it when to do something) by `$emit`ing close and open events when associated DOM events are clicked (since these DOM events are the only way a user can interact with a modal while it's open). We do a little bit of this[ in Kolibri](https://github.com/learningequality/kolibri/blob/develop/kolibri/core/assets/src/views/KModal.vue#L231) for the "close" options presented to a user within the modal. The parent listens for the event, does what it has to do, and flips the boolean controlling the modal's visibility. 

IMO, it all kind of depends on the use case. And whether or not you want a route associated with the modal.

> There are also event listeners like clicking outside of the modal or pressing the escape button that trigger closing as well.
[We added the listeners ourselves](https://github.com/learningequality/kolibri/blob/develop/kolibri/core/assets/src/views/KModal.vue#L9), as Vue makes it pretty easy, but it's admittedly less boilerplate (The first implementation of the modal also bound a click listener that was nearly identical, but that was [removed down the line](https://github.com/learningequality/kolibri/blob/ae1dd6b8a6d27ba967726811895011925f133cb5/kolibri/core/assets/src/views/core-modal/index.vue#L8)). 

I'm with Richard regarding JQuery, but you probably know best regarding how long it'd take to develop!

Reply:
I'll add this to tech debt for now as this sounds like it will take some time to implement